distributed/protocol/serialize.py

```diff
@@ -69,19 +69,21 @@ def pickle_dumps(x, context=None):
 
 def pickle_loads(header, frames):
     x, buffers = frames[0], frames[1:]
-    writeable = header["writeable"]
-    for i in range(len(buffers)):
-        mv = memoryview(buffers[i])
-        if writeable[i] == mv.readonly:
+
+    new = []
+    memoryviews = map(memoryview, buffers)
+    for writeable, mv in zip(header["writeable"], memoryviews):
```

Comment:
This would be an issue if `header["writeable"]` is `()`, which came up in another issue

Reply:
This PR is focused on removing the list mutation.  I recommend that we handle the `not header["writeable"]` issue separately if that's ok.